### PR TITLE
Fix failing Command test cases

### PR DESCRIPTION
**Pull request by Lemonsr**
Tests fixed:
- AddCommandIntegrationTest 
- AddCommandTest
- ClearCommandTest 
- DeleteCommandTest 
- EditCommandTest 
- ListCommandTest 
- EditStudentDescriptorTest 
- FindCommandTest 
- LogicManagerTest 

> ❗  Note: we should split up tutor and student tests into separate test cases for better maintainability and reduce occurrences of bugs in tests.

Resolves some test cases from #59

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/DeleteCommandTest.java [hunk 1]
```diff
@@ -30,86 +30,98 @@
...
-        DeleteCommand deleteCommandStudent = new DeleteCommand(outOfBoundIndexStudent, PersonType.STUDENT);
 
-        // Fail at deleting a tutor for invalid index for an unfiltered list
         assertCommandFailure(deleteCommandTutor, model, Messages.MESSAGE_INVALID_TUTOR_DISPLAYED_INDEX);
-        // Fail at deleting a student for invalid index for an unfiltered list
+    }
+
+    @Test
+    public void execute_invalidIndexUnfilteredListStudent_throwsCommandException() {
+        Index outOfBoundIndexStudent = Index.fromOneBased(model.getFilteredStudentList().size() + 1);
```
Comment (zihaooo9) [inline src/test/java/seedu/address/logic/commands/DeleteCommandTest.java:64]: You can make use of the `INVALID_INDEX` in the CommandTestUtil.java file for this.

Comment (Lemonsr) [reply src/test/java/seedu/address/logic/commands/DeleteCommandTest.java:64]: `INVALID_INDEX` is a String, but we want to use an Index type here, so I don't think I will change this

File: src/test/java/seedu/address/logic/commands/AddCommandIntegrationTest.java [hunk 2]
```diff
@@ -29,28 +29,39 @@ public void setUp() {
...
+        Model expectedModelStudent = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModelStudent.addStudent(validStudent);
+
+        assertCommandSuccess(new AddCommand(validStudent, PersonType.STUDENT), model,
+                String.format(AddCommand.MESSAGE_SUCCESS_STUDENT, validStudent), expectedModelStudent);
+    }
+
+    @Test
+    public void execute_duplicateTutor_throwsCommandException() {
         Tutor tutorInList = model.getAddressBook().getTutorList().get(0);
```
Comment (kleonang) [inline src/test/java/seedu/address/logic/commands/AddCommandIntegrationTest.java:55]: Can make use of predefined `INDEX_FIRST_PERSON` here:
```suggestion
        Tutor tutorInList = model.getAddressBook().getTutorList().get(INDEX_FIRST_PERSON.getZeroBased());
```

File: src/test/java/seedu/address/logic/commands/AddCommandIntegrationTest.java [hunk 3]
```diff
@@ -29,28 +29,39 @@ public void setUp() {
...
-        Student studentInList = model.getAddressBook().getStudentList().get(0);
 
         assertCommandFailure(new AddCommand(tutorInList, PersonType.TUTOR),
                 model, AddCommand.MESSAGE_DUPLICATE_TUTOR);
+    }
+
+    @Test
+    public void execute_duplicateStudent_throwsCommandException() {
+        Student studentInList = model.getAddressBook().getStudentList().get(0);
```
Comment (kleonang) [inline src/test/java/seedu/address/logic/commands/AddCommandIntegrationTest.java:63]: Same as above, can make use of predefined `INDEX_FIRST_PERSON` to reduce magic numbers.

File: src/test/java/seedu/address/stubs/FindCommandParserStub.java [hunk 4]
```diff
@@ -17,6 +18,7 @@
      */
     public static Name extractName(String args) throws ParseException {
```
Comment (kleonang) [inline src/test/java/seedu/address/stubs/FindCommandParserStub.java:19]: There may not be any usages of this method after the current changes, may not need `FindCommandParserStub.java` anymore if `extractName()` is not used. But ok to leave in if there may be future tests that require it.

**Review by kleonang — APPROVED** [review]

**Pull request merged**